fvm/src/kernel/default.rs

```diff
@@ -295,8 +307,14 @@ where
             // to be in the state-tree.
             .or_fatal()?;
 
-        // We charge on open, not read, to emulate the current gas model.
         let block = Block::new(cid.codec(), data);
+
+        self.call_manager.charge_gas(
+            self.call_manager
+                .price_list()
+                .on_block_open_per_byte(block.size().try_into().or_illegal_argument()?),
```

Comment:
You know, this is a case where I'd just cast. The block size limit will be well below this.